### PR TITLE
SUBMARINE-690. Enable pre-defined web interface in submarine workbench

### PR DESCRIPTION
**Pull request by ByronHsu**
### What is this PR for?

**Add pre-defined web interface.**

The steps for users to create a pre-defined experiment are as following:
1. Click pre-defined button
2. Pre-defined data will be fetched from db
3. Render configurable params and spec on ng-modal
4. When the form is validated, the experiment can be submitted.

### What type of PR is it?
[Feature]

### Todos
- [ ] Test predefined component (The e2e test failed now due to unknown reason)
- [ ] Real-time fetch experiment status
- [ ] The implementation logic of the experiment page should be improved (e.g. split components correctly, use resolver to avoid long-time waiting of data)

### What is the Jira issue?
https://issues.apache.org/jira/projects/SUBMARINE/issues/SUBMARINE-690

### How should this be tested?
https://github.com/ByronHsu/submarine/actions/runs/404158134

### Screenshots (if appropriate)
Kapture 2020-12-07 at 17 18 34

### CODE REVIEW

**Comment by ByronHsu** [comment]
@tangzhankun @pingsutw @xunliu 
Please help me to review this PR. Thanks!

**Comment by pingsutw** [comment]
I will review as soon as possible~

File: submarine-workbench/workbench-web/src/app/pages/workbench/experiment/experiment-home/experiment-home.component.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@
  * "License"); you may not use this file except in compliance
  * with the License.  You may obtain a copy of the License at
  *
- *   http://www.apache.org/licenses/LICENSE-2.0
+ *   http: //www.apache.org/licenses/LICENSE-2.0
```
Comment (tangzhankun) [inline submarine-workbench/workbench-web/src/app/pages/workbench/experiment/experiment-home/experiment-home.component.ts:10]: This space is not needed actually. Maybe some settings in your IDE cause this?

**Review by xunliu — COMMENTED** [review]
@ByronHsu Please change PR title to `SUBMARINE-690. xxxx`

File: submarine-workbench/workbench-web/src/app/pages/workbench/experiment/experiment-home/experiment-home.component.ts (hunk 1 above)
Comment (ByronHsu) [reply submarine-workbench/workbench-web/src/app/pages/workbench/experiment/experiment-home/experiment-home.component.ts]: Thanks. I've fixed it.

File: submarine-workbench/workbench-web/src/app/pages/workbench/experiment/experiment-predefined-form/experiment-predefined-form.component.ts [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    private fb: FormBuilder,
+    private nzMessageService: NzMessageService
+  ) {}
+
+  ngOnInit() {
+    this.experimentService.fetchExperimentTemplateList().subscribe((res) => {
+      this.templates = this.parseTemplateRespond(res);
+
+      if (Object.keys(this.templates).length != 0) {
+        // default: switch to option 1
```
Comment (pingsutw) [inline submarine-workbench/workbench-web/src/app/pages/workbench/experiment/experiment-predefined-form/experiment-predefined-form.component.ts:73]:
```suggestion
        // default: switch to first template
```

File: submarine-workbench/workbench-web/src/app/pages/workbench/experiment/experiment.component.ts [hunk 3]
```diff
@@ -101,7 +101,7 @@ export class ExperimentComponent implements OnInit {
       this.modalProps = { ...this.modalProps, ...props };
     });
 
-    this.reloadCheck();
+    // this.reloadCheck();
```
Comment (pingsutw) [inline submarine-workbench/workbench-web/src/app/pages/workbench/experiment/experiment.component.ts:104]: Why comment out this line?

**Review by pingsutw — COMMENTED** [review]
Overall LGTM, Thanks @ByronHsu 
@kobe860219 could you also take a look?

File: submarine-workbench/workbench-web/src/app/pages/workbench/experiment/experiment.component.ts (hunk 3 above)
Comment (ByronHsu) [reply submarine-workbench/workbench-web/src/app/pages/workbench/experiment/experiment.component.ts:104]: Oh, I accidentally commented on this line. 
However, I think there is a better way to handle this issue. I will check it later when I start to refactor the code.

**Review by xunliu — APPROVED** [review]
@ByronHsu Thank for you contribution. 
LGTM, will merge if no more comments.

**Pull request closed without merging**